flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -246,6 +244,21 @@ public static String getTaskManagerShellCommand(
         return startCommand;
     }
 
+    public static String generateJvmOptsString(
```

Comment:
nit: It's better to add a simple unit test for this method before merging.

BTW, you can squash all commits directly after adding the unit test.